qml-ts-mode.el

```diff
@@ -181,138 +183,262 @@ See `treesit-thing-settings' for more information."))
   "QML keywords for tree-sitter font-locking.")
 
 
-(defvar qml--treesit-font-lock-settings
-  (treesit-font-lock-rules
+(defun qml--is-node-id? (node)
+  "Check if NODE is an ui_binding node corresponding to the id."
+  (string= (treesit-node-text node) "id"))
 
-   :language 'qmljs
-   :feature 'comment
-   '((comment) @font-lock-comment-face)
 
-   :language 'qmljs
-   :feature 'constant
-   '(((identifier) @font-lock-constant-face
-      (:match "^[A-Z_][A-Z_\\d]*$" @font-lock-constant-face))
-     [(true) (false) (null)] @font-lock-constant-face)
+(defconst qml--treesit-font-lock-comment-settings
+  '((comment) @font-lock-comment-face))
 
-   :language 'qmljs
-   :feature 'keyword
-   `([,@js--treesit-keywords] @font-lock-keyword-face
-     [(this) (super)] @font-lock-keyword-face
-     [,@qml--treesit-keywords] @font-lock-keyword-face)
 
-   :language 'qmljs
-   :feature 'string
-   '((regex pattern: (regex_pattern)) @font-lock-regexp-face
-     (string) @font-lock-string-face)
+(defconst qml--treesit-font-lock-definitions-settings
+  '((function_declaration
+     name: (identifier) @font-lock-function-name-face)
 
-   :language 'qmljs
-   :feature 'string-interpolation
-   :override t
-   '((template_string) @js--fontify-template-string
-     (template_substitution ["${" "}"] @font-lock-misc-punctuation-face))
+    (variable_declarator
+     name: (identifier) @font-lock-variable-name-face)
 
-   :language 'qmljs
-   :feature 'type
-   '((type_identifier) @font-lock-type-face
+    (ui_property
+     name: (identifier) @font-lock-property-name-face)
 
-     (predefined_type) @font-lock-type-face
+    ;;   (variable_declarator
+    ;;    name: (array_pattern
+    ;;           (identifier)
+    ;;           (identifier)
+    ;;           @font-lock-function-name-face)
+    ;;    value: (array (number) (function)))
 
-     (ui_object_definition
-      type_name: (identifier) @font-lock-type-face)
+    (import_clause (identifier) @font-lock-variable-name-face)
+    (import_clause (named_imports (import_specifier (identifier))
+                                  @font-lock-variable-name-face))
 
-     (ui_object_definition
-      type_name: (nested_identifier) @font-lock-type-face))
+    (required_parameter (identifier) @font-lock-variable-name-face)
 
-   :language 'qmljs
-   :feature 'definition
-   '((function_declaration
-      name: (identifier) @font-lock-function-name-face)
+    (optional_parameter (identifier) @font-lock-variable-name-face)
+
+    (ui_signal
+     name: (identifier) @font-lock-function-name-face)
+
+    (ui_binding
+     name: [
+            (identifier) @font-lock-variable-use-face
+
+            ;; I'm not sure if this one ever appears
+            (nested_identifier) @font-lock-variable-name-face
```

Comment:
I did not find this in the AST in the QML file I was using to test things. It might be something that does not exist in the current grammar, or just a construct that I forgot or that I'm not aware of. Since it's not breaking anything, I had to reason to remove it and cause a regression in the font-locking.